pkg/client/cache/controller.go

```diff
@@ -61,21 +61,23 @@ type Config struct {
 type ProcessFunc func(obj interface{}) error
 
 // Controller is a generic controller framework.
-type Controller struct {
+type controller struct {
 	config         Config
 	reflector      *Reflector
 	reflectorMutex sync.RWMutex
 }
 
 // TODO make the "Controller" private, and convert all references to use ControllerInterface instead
-type ControllerInterface interface {
+type Controller interface {
 	Run(stopCh <-chan struct{})
 	HasSynced() bool
+	LastSyncResourceVersion() string
+	Requeue(obj interface{}) error
```

Comment:
> Requeue takes ownership of the underlying reflector queue in a naughty way.

@deads2k, What's naughty about it? We don't take its ownership, we just need to inject an object there.

> claims from cache.Store to cache.Queue

No, that a different cache and it cannot be a queue.

Without `.Requeue` we would need to create our own worker queue and goroutines in PV controller. Sure, it's not that complicated, some controllers have their own worker queues. On the other hand, PV controller is already pretty complicated enough and this would add another layer of complexity.